threads-csharp/PeriodicTimerCSharp/PeriodicTimerCSharp/PeriodicTimerMethods.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace PeriodicTimerCSharp;
+
+public class PeriodicTimerMethods
+{
+    private readonly PeriodicTimer _periodicTimer;
+    private readonly CancellationTokenSource _cancellationToken = new();
+    private Task? _task;
+    private int _size;
+
+    public PeriodicTimerMethods(TimeSpan timeSpan, int size) 
+    {
+        _periodicTimer = new PeriodicTimer(timeSpan);
+        _size = size;
+    }
+
+    public static int[] CreateRandomArray(int size)
+    {
+        var array = new int[size];
+
+        for (int i = 0; i < size; i++)
+        {
+            array[i] = Random.Shared.Next();
+        }
+        
+        return array;
+    }
+
+    public void Start() 
+    {
+        _task = RandomArrayTimerAsync();
+    }
+
+    private async Task RandomArrayTimerAsync() 
+    {
+        try 
+        {
+            var stopWatch = System.Diagnostics.Stopwatch.StartNew();
+            var numberOfArrays = 1;
+
+            while (await _periodicTimer.WaitForNextTickAsync(_cancellationToken.Token)) 
+            {
+                var array = CreateRandomArray(_size);
+
+                foreach (var item in array)
+                {
+                    Console.WriteLine(item);
+                }
+
+                Console.WriteLine($"Elapsed time: {stopWatch.Elapsed.Seconds}s" +
+                    $" and {numberOfArrays++} arrays created");
```

Comment:
I am aware that you split that into two different rows with a `+` string concatenation but I don't think this is a good solution. We shouldn't compromise our code best practices because of the restrictions of our article. That said, it would be better to make the wording smaller instead of performing this type of concatenation.

Let's do it like this:
`Console.Writeline($"Created {numberOfArrays++} in {stopWatch.Elapsed.Seconds}s")`

In that way, we won't compromise our code best practices and we will have a shorter output 😄

Reply:
Hi @mahrncic, that's a great idea, I have edited the code and article.